src/plans/stepsize.jl

```diff
@@ -660,30 +660,614 @@ If ``$(_tex(:norm, "X_k"; index = "p_k")) ≤ αω_{k-1}``, then let
+``p'(a.t)=a.df``, ``p'(b.t)=b.df``.
+
+# Input
+* `a::UnivariateTriple{R}`: triple of bracket value `a`
+* `b::UnivariateTriple{R}`: triple bracket value `b`
+"""
+function secant(a::UnivariateTriple{R}, b::UnivariateTriple{R}) where {R}
+    return (a.t * b.df - b.t * a.df) / (b.df - a.df)
+end
+
+"""
+    cubic_stepsize_update_step(a::Real, b::Real, c::Real, τ::Real)
+
+
+Step function to determine the stepsize update `c` described in
+[Hager:1989](@cite).
+
+# Input
+* `a::Real`: first value of the bracket
+* `b::Real`: second value of the bracket
+* `c::Real`: update value
+* `τ::Real`: minimal step tolerance
+"""
+function cubic_stepsize_update_step(a::Real, b::Real, c::Real, τ::Real)
+    y = min(a, b)
+    z = max(a, b)
+    if (y + τ ≤ c && c ≤ z - τ)
+        return c
+    end
+    if (c > (a + b) / 2)
+        return max(z - τ, (a + b) / 2)
+    else
+        return min(y + τ, (a + b) / 2)
+    end
+end
+
+"""
+Get the `UnivariateTriple` of the problem `mp` related to the step with
+stepsize ``t`` from ``p`` in direction ``η``.
+
+# Input
+* `mp::AbstractManoptProblem`
+* `cbls:::CubicBracketingLinesearchStepsize`: containing `retraction_method`, `vector_transport` and the temporary `candidate_point` and `candidate_direction`
+* `p`: point in the manifold of `mp`
+* `η`: search direction at `p`
+* `t::Real`: step size 
+"""
+function get_univariate_triple!(mp::AbstractManoptProblem, cbls::CubicBracketingLinesearchStepsize, p, η, t)
+    M = get_manifold(mp)
+    cbls.last_stepsize = t
+    ManifoldsBase.retract_fused!(M, cbls.candidate_point, p, η, t, cbls.retraction_method)
+    vector_transport_to!(M, cbls.candidate_direction, p, η, cbls.candidate_point, cbls.vector_transport_method)
+    f, df = get_cost_and_differential(mp, cbls.candidate_point, cbls.candidate_direction)
+    return UnivariateTriple(t, f, df)
+end
+
+function (cbls::CubicBracketingLinesearchStepsize)(
+        mp::AbstractManoptProblem,
+        s::AbstractManoptSolverState,
+        k::Int,
+        η = (-get_gradient(mp, get_iterate(s)));
+        kwargs...,
+    )
+    M = get_manifold(mp)
+    p = get_iterate(s)
+
+    init = UnivariateTriple(0.0, get_cost(M, get_objective(mp), p), get_differential(mp, p, η; gradient = s.X, evaluated = true))
+
+    check_curvature(c::UnivariateTriple) = abs(c.df) < cbls.sufficient_curvature * abs(init.df)
+
+    n_iter = 0
+    t = cbls.last_stepsize
+    c_old = init
+    c = get_univariate_triple!(mp, cbls, p, η, t)
+    a, b = nothing, nothing
+    # Construct initial bracket
+    while ((n_iter += 1) <= cbls.max_iterations)
+        (c.f < init.f && check_curvature(c)) && return t
+        if (c.f ≥ c_old.f && c_old.df * (c.t - c_old.t) < 0)
+            (a, b) = c_old, c
+            break
+        end
+        if (c.f ≤ c_old.f && c.df * (c_old.t - c.t) < 0)
+            (a, b) = c, c_old
+            break
+        end
+        (t == cbls.max_stepsize) && return t
```

Comment:
Hi @jonas-pueschel ! I'm revisiting this stepsize to add proper direction-specific maximum stepsizes and this line seems odd to me. What's the justification for stopping the search here? I'd expect to proceed to the next loop (break instead of return) if we can no longer expand instead.